cmd/occollector/app/builder/receivers_builder_test.go

```diff
@@ -0,0 +1,257 @@
+
+	// Ensure receiver has its fields correctly populated.
+	require.NotNil(t, receiver)
+
+	if test.hasTraces {
+		assert.NotNil(t, receiver.trace)
+	} else {
+		assert.Nil(t, receiver.trace)
+	}
+
+	if test.hasMetrics {
+		assert.NotNil(t, receiver.metrics)
+	} else {
+		assert.Nil(t, receiver.metrics)
+	}
+
+	// Compose the list of created exporters.
+	var exporters []*builtExporter
+	for _, name := range test.exporterNames {
+		// Ensure exporter is created.
+		exp := allExporters[config.Exporters[name]]
+		require.NotNil(t, exp)
+		exporters = append(exporters, exp)
+	}
+
+	// Send TraceData via receiver and verify that all exporters of the pipeline receive it.
+
+	// First check that there are no traces in the exporters yet.
+	for _, exporter := range exporters {
+		consumer := exporter.tc.(*configv2.ExampleExporterConsumer)
+		require.Equal(t, len(consumer.Traces), 0)
+		require.Equal(t, len(consumer.Metrics), 0)
+	}
+
+	// Send one trace.
+	name := tracepb.TruncatableString{Value: "testspanname"}
+	traceData := data.TraceData{
+		SourceFormat: "test-source-format",
+		Spans: []*tracepb.Span{
+			{Name: &name},
+		},
+	}
+	if test.hasTraces {
+		traceProducer := receiver.trace.(*configv2.ExampleReceiverProducer)
+		traceProducer.TraceConsumer.ConsumeTraceData(context.Background(), traceData)
+	}
+
+	metricsData := data.MetricsData{
+		Metrics: []*metricspb.Metric{
+			{MetricDescriptor: &metricspb.MetricDescriptor{Name: "testmetric"}},
+		},
+	}
+	if test.hasMetrics {
+		metricsProducer := receiver.metrics.(*configv2.ExampleReceiverProducer)
+		metricsProducer.MetricsConsumer.ConsumeMetricsData(context.Background(), metricsData)
+	}
+
+	// Now verify received data.
+	for _, name := range test.exporterNames {
+		// Check that the data is received by exporter.
+		exporter := allExporters[config.Exporters[name]]
+
+		// Validate traces.
+		if test.hasTraces {
+			var spanDuplicationCount int
+			if test.spanDuplicationByExporter != nil {
+				spanDuplicationCount = test.spanDuplicationByExporter[name]
+			} else {
+				spanDuplicationCount = 1
+			}
+
+			traceConsumer := exporter.tc.(*configv2.ExampleExporterConsumer)
+			require.Equal(t, spanDuplicationCount, len(traceConsumer.Traces))
+
+			for i := 0; i < spanDuplicationCount; i++ {
+				assert.Equal(t, traceData, traceConsumer.Traces[i])
+
+				// Check that the span was processed by "attributes" processor and an
+				// attribute was added.
+				assert.Equal(t, int64(12345),
+					traceConsumer.Traces[i].Spans[0].Attributes.AttributeMap["attr1"].GetIntValue())
+			}
+		}
+
+		// Validate metrics.
+		if test.hasMetrics {
+			metricsConsumer := exporter.mc.(*configv2.ExampleExporterConsumer)
+			require.Equal(t, 1, len(metricsConsumer.Metrics))
+			assert.Equal(t, metricsData, metricsConsumer.Metrics[0])
+		}
+	}
+}
+
+func TestReceiversBuilder_Error(t *testing.T) {
+	//config, err := configv2.LoadConfigFile(t, "testdata/pipelines_builder.yaml")
```

Comment:
Why this test with content commented out?

Reply:
Forgot to uncomment and fix. Doing now.

Reply:
Fixed. Please check TestReceiversBuilder_DataTypeError()